### DISCUSSION TITLE
Add BitDepth to TiffMetadata

### DISCUSSION THREAD

**Discussion by PeteMichaud** [SixLabors/ImageSharp#2053] [Ideas] [answered]
Per the discussion here: https://github.com/SixLabors/ImageSharp/issues/2049

`TiffMetadata` doesn't contain bit depth information because Tiff files can contain multiple frames, which may have different bit depths.

It seems like an area for possible API improvement though.

Consider that different tiff frames can be different sizes too, yet a tiff loaded as an `Image` still has Width and Height fields. Consider also that it would be very confusing if it didn't have those fields. 

I didn't look at the implementation here, but I assume it's either the Width and Height from the first frame or the Max Width and Height from all frames. I am guessing that empirically most of the time when a consumer loads a tiff file it only has one frame, and will probably want to deal with it as one frame without thinking of the format details of tiff. When a consumer is using multiple frames deliberately, they will have access as usual to all that frame information.

So along the same lines as Width and Height, I am proposing some way of exposing the BitDepth information for a tiff using the same facilities as one might use for a png file, for example (ie. exposed on the `TiffMetadata` object). I could imagine it working in a couple different ways, or maybe both of these:

1. The metadata for the image contains a `TiffBitDepth` field  that is either the bit depth for the first frame, or the maximum bit depth of all the frames, depending how width and height are populated (probably you want to do it whatever that way was, for consistency). This would handle what I assume is the default use case in an identical way to other file types. 
  * This would solve my use case where I store the bit depth when I load the file so that after processing, I can save the new file in a matching bit depth rather than having to save it at the max bit depth supported by the filetype just in case.
  * To be more concrete, my user passes in an image of a planet map, my software performs a map projection operation (like converting it to Mercator), then returns the new image with the projected map. If my user passes me a 16bit png, I want to pass back a 16bit png. If my user passes me a 32bit tiff, I want to pass back a 32bit tiff.  
3. The `TiffMetadata` (and I guess the metadata for any image type that potentially has multiple frames) has a frame metadata collection on it, containing all the metadata you might get from a `image.Frames[frameIdx].Metadata.GetTiffMetadata()` loop. This would make the API way more discoverable. I am not aware of any way to have discovered this frame meta data in the api or the docs prior to raising the issue I raised and getting an answer from Joe about it.

**Comment by JimBobSquarePants** [comment, marked as answer]
See #1982 regarding Width and Height properties of Tiff images. 

As I described [previously](https://github.com/SixLabors/ImageSharp/issues/2049#issuecomment-1062517052) Tiff is a container format where different frames can have different bit depths. This rendered a global property functionally redundant only adding unnecessary complexity to the encoding codebase..
